From c91534a920970413e0e2aea543bb1bef43b83b94 Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:38:26 -0500
Subject: [drm] radeon: tweak ACCEL_WORKING2 query for hawaii

Message-id: <1418852380-13061-16-git-send-email-rclark@redhat.com>
Patchwork-id: 102153
O-Subject: [RHEL7 drm 15/89] drm/radeon: tweak ACCEL_WORKING2 query for hawaii
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: since v3.16.4

commit af372afef3b8fb41b17c926f4b85de678ad97591

Author:     Alex Deucher <alexander.deucher@amd.com>
AuthorDate: Fri Aug 1 20:05:30 2014 +0200
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:10:43 2014 -0500

    drm/radeon: tweak ACCEL_WORKING2 query for hawaii

    commit 3c64bd26f7e9bd589ebe0d1ebec69ef2f784c12d upstream.

    Return 2 so we can be sure the kernel has the necessary
    changes for acceleration to work.

    Note: This patch depends on these two commits:
     - drm/radeon: fix cut and paste issue for hawaii.
     - drm/radeon: use packet2 for nop on hawaii with old firmware

    Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
    Signed-off-by: Andreas Boll <andreas.boll.dev@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Rob Clark <rclark@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/radeon/radeon_kms.c b/drivers/gpu/drm/radeon/radeon_kms.c
index d38bcb6..c1a206d 100644
--- a/drivers/gpu/drm/radeon/radeon_kms.c
+++ b/drivers/gpu/drm/radeon/radeon_kms.c
@@ -254,10 +254,14 @@ static int radeon_info_ioctl(struct drm_device *dev, void *data, struct drm_file
   }
   break;
  case RADEON_INFO_ACCEL_WORKING2:
-  if (rdev->family == CHIP_HAWAII)
-   *value = (!!rdev->accel_working) * 2;
-  else
+  if (rdev->family == CHIP_HAWAII) {
+   if (rdev->accel_working)
+    *value = 2;
+   else
+    *value = 0;
+  } else {
    *value = rdev->accel_working;
+  }
   break;
  case RADEON_INFO_TILING_CONFIG:
   if (rdev->family >= CHIP_BONAIRE)
-- 
1.7.1